photosynthesis_metrics/vif.py

```diff
@@ -51,7 +52,13 @@ def vif_p(prediction: torch.Tensor, target: torch.Tensor,
         See https://live.ece.utexas.edu/research/Quality/VIF.htm for details.
         
     """
-    assert prediction.dim() == 4, f'Expected 4D tensor, got {prediction.size()}'
+    _validate_input((prediction, target), allow_5d=False)
+    prediction, target = _adjust_dimensions(input_tensors=(prediction, target))
+
+    min_size = 41
```

Comment:
Please, add comment how to get this number.

Reply:
That's magic 🧙 
Due to progressive downsampling and different kernel sizes on different levels I didn't find formula to correctly estimate min size. I just tried a few different resolutions